### PR TITLE
Damon refactor ratio stats mki

### PR DESCRIPTION
**Pull request by Damonamajor**
This adds MKI into the existing ratio stats. It also deletes to code for the base functions as well as the _met and _ci functions to directly access assesspy.

- things like cod_boot are included in the assesspy package, but can't be accessed from the package to use outside of it. 
- The outputs are still set to the test folder. It needs to be shifted back to the original folder. 
- I've been having issues removing the ap. before the functions since it also renames the column names. Because of that they are imported separately. I have a structure in place and am working through it, but it's just time consuming running checks on it in AWS. I'll send a separate push if this is necessary to fix.

### CODE REVIEW

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [inline aws-glue/jobs/reporting-ratio_stats.py:53]: **[Nitpick]** In the PR body, you mentioned that the outputs should be reset to the prod folder:

> The outputs are still set to the test folder. It needs to be shifted back to the original folder.

I agree with this, but I think that now that we're proposing an official change to the script via this pull request, there's no need for the test bucket anymore, so we might as well return to using `s3_ratio_stats` now:

```suggestion
```

If we need to test any further changes, we can always switch it back!

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [inline aws-glue/jobs/reporting-ratio_stats.py:431]: **[Nitpick]** Same as above, let's change this back to the production bucket:

```suggestion
).to_parquet(s3_ratio_stats)
```

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [inline aws-glue/jobs/reporting-ratio_stats.py:12]: **[Nitpick]** A couple notes here about imports:

1. The `as` keyword is only necessary if we want to create an alias for the package we're importing (e.g. `import assesspy as ap`). Since in these cases we're not changing the name of the packages we're importing, we don't need to include aliases.
2. If we want to import multiple symbols from one package, we can consolidate them into one `import` statement by separating the symbols with commas, e.g.:

```python
from assesspy import cod_ci, boot_ci, prb, ...
```

3. The symbol imports on lines 1-11 are duplicative with the `import assesspy as ap` call on line 12, and I can't tell if the import on line 12 is being used anywhere. We should either import these symbols from the package, or import the package directly, but not both.

Putting this all together, my suggestion would be:

```suggestion
from assesspy import (
    boot_ci,
    cod,
    cod_ci,
    cod_met,
    detect_chasing,
    mki,
    mki_met,
    prb,
    prb_met,
    prd
    prd_met,
)
```

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [inline aws-glue/jobs/reporting-ratio_stats.py:36]: **[Nitpick, non-blocking]** Looks like this comment is no longer in the right place? But I also think it was never really necessary to begin with, we should expect the reader to know what an `import` is for.

```suggestion
```

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [inline aws-glue/jobs/reporting-ratio_stats.py:167]: **[Question, non-blocking]** You mentioned in the PR body that this function is defined in `assesspy`, but isn't available for use by package consumers:

> things like cod_boot are included in the assesspy package, but can't be accessed from the package to use outside of it.

 Can you point me to the lines in `assesspy` where it's defined? I couldn't find it, but if it's true that it's defined in `assesspy`, we should try to figure out how to make it available to this script too.

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [reply aws-glue/jobs/reporting-ratio_stats.py:12]: Reading the PR body again, it seems like this might be a result of difficulty refactoring to use the `assesspy` import:

> I've been having issues removing the ap. before the functions since it also renames the column names. Because of that they are imported separately. I have a structure in place and am working through it, but it's just time consuming running checks on it in AWS. I'll send a separate push if this is necessary to fix.

I don't quite understand what the trouble is, perhaps we could pair on it together?

**Review by jeancochrane — CHANGES_REQUESTED** [review]
Looking great so far! Just a few nitpicks below. I'm not totally sure why `black` started failing on CI, seeing as it seems like the formatting it was complaining about pre-dated your changes; but I think it's good to switch to `black` formatting for this script anyway so we might as well do it now.

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (jeancochrane) [inline aws-glue/jobs/reporting-ratio_stats.py:11]: **[Nitpick, non-blocking]** We could collapse this all into one import

```suggestion
from assesspy import (
    boot_ci,
    cod,
    cod_ci as cod_boot,
    cod_met,
    detect_chasing,
    mki,
    mki_met,
    prb,
    prb_met,
    prd,
    prd_ci as prd_boot,
    prd_met
)
```

**Review by jeancochrane — APPROVED** [review]
Looks great to me! Let's get functional review from @dfsnow before we merge 👍🏻

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (dfsnow) [reply aws-glue/jobs/reporting-ratio_stats.py:11]: It turns out that `isort` really hates the more pythonic syntax, so sticking with Damon's original imports.

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (dfsnow) [inline aws-glue/jobs/reporting-ratio_stats.py:140]: @wrridgeway This is copied from the original code, but shouldn't this line be:

```suggestion
    median_ci = median_boot(x_no_outliers, nboot=1000)
```

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (dfsnow) [inline aws-glue/jobs/reporting-ratio_stats.py:174]: **nitpick:** Drop this unused docstring.

**Review by dfsnow — COMMENTED** [review]
Nice work @Damonamajor! This is nearly good to go, see my few comments, one of which is for @wrridgeway.

@wrridgeway Can you give this a final PR review so we can merge it this week?

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (wrridgeway) [reply aws-glue/jobs/reporting-ratio_stats.py:140]: Yes, it should be.

**Comment by wrridgeway** [comment]
Could we run the new and the old script and compare their output if this hasn't already been done?

**Comment by Damonamajor** [comment]
> @Damonamajor Could we run the new and the old script and compare their output if this hasn't already been done?

Yup, I'll do it after this meeting.

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (Damonamajor) [reply aws-glue/jobs/reporting-ratio_stats.py]: @wrridgeway @dfsnow I'm not sure why, but this change seems to make it not run on AWS.

**Comment by Damonamajor** [comment]
> > @Damonamajor Could we run the new and the old script and compare their output if this hasn't already been done?
> 
> Yup, I'll do it after this meeting.

I ran and downloaded the files. The dimensions are the same (aside from adding mki things), as well as the outputs. The one exception is the confidence intervals, which are very minorly off. Billy said this was a result of bootstrapping.

**Review by wrridgeway — CHANGES_REQUESTED** [review]

File: aws-glue/jobs/reporting-ratio_stats.py
Comment (Damonamajor) [reply aws-glue/jobs/reporting-ratio_stats.py:6]: @wrridgeway The tables are named along with the functions. Doing this, without extra steps, would result in title names being ap.cod_ci.

**Review by wrridgeway — APPROVED** [review]

**Pull request merged**
